Fe#14 data selection

stripes416 opened the pull request.

stripes416 committed:
Add frontend to endpoint retrieval of JSON object -- Leaf wetness

Once JSON object is retrieved, user is directed to
  http://localhost:8000/env_data?vineyard_id=0&env_variable=leafwetness
which displays the JSON object.

Any errors in data retrieval direct the user back to
  dashboard.html

stripes416 committed:
Add id attribute to env_cond buttons, modify .js to use ids

Adds id attributes to the environmental condition buttons on the map
overlay in dashboard.html.

Also modifies the leaf wetness selection to use the id attribute instead
of the title in dashboard.js.

stripes416 committed:
Add frontend to endpoint retrieval of JSON object -- temperature

Once JSON object is retrieved, user is directed to
  http://localhost:8000/env_data?vineyard_id=0&env_variable=temperature
which displays the JSON object.

Any errors in data retrieval direct the user back to
  dashboard.html

stripes416 committed:
Add frontend to endpoint retrieval of JSON object -- humidity

Once JSON object is retrieved, user is directed to
  http://localhost:8000/env_data?vineyard_id=0&env_variable=humidity
which displays the JSON object.

Any errors in data retrieval direct the user back to
  dashboard.html

stripes416 committed:
Add ids to the env_cond selections in the menu

Adds ids to each of the environmental conditions in the pop-up menu.

stripes416 committed:
Add env_cond selection for the pop-up menu options

Has the same results as the map overlay buttons

something15525 commented:
@stripes416 is this supposed to just show the data values and get rid of the map when items are selected?

something15525 commented on js/dashboard.js:
```diff
@@ -199,6 +199,110 @@ $(function() {
+            if (json.env_data) {
+                window.location.href = "http://localhost:8000/env_data?vineyard_id=0&env_variable=leafwetness";
+            } else {
+                // go nowhere
```
These "go nowhere" statements seem unnecessary. shouldn't we just do nothing at all? or is there a reason we're specifying a window location to no where?

stripes416 replied:
I didn't know what to do. Is a pop up error message appropriate here? Should we do nothing at all?

stripes416 replied:
Okay, after coffee, I'm thinking more clearly. I added those as basic placeholders. Didn't we say in the design phase that we would error handle? Depending on when the error occurs, shouldn't we have it go to a basic "here's the thing that went wrong, sorry. Try to reload the page in a few minutes" page?

something15525 replied:
But we're not including that for MVP, right? I don't think placeholders are necessary right now. And if we do leave them in there, we should add a TODO line.

something15525 commented on js/.idea/vcs.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Personal opinion, but the .idea folder should be completely ignored.

stripes416 replied:
Sure

something15525 replied:
Okay, I'll add a commit removing them.

stripes416 commented:
Again, I didn't know what to do. The next step is for someone to take the data that's retrieved and actually generate the map with it.

something15525 committed:
Add .idea to .gitignore, and remove .idea folder from js folder

* Added js/.idea as well, just in case

something15525 commented:
The only change I would still make is to add TODOs to the areas that "go nowhere", then I think it should be good to merge!

stripes416 committed:
Add TODO: at each point of possible error  

Per suggestions by @michael.limb, the //go nowhere messages have been changed to a TODO:, indicating a need for error handling, of which type should be decided on by the team.

scotte216 commented:
It works for me: Clicking the various icons will display the appropriate data in text form.

 I think having error handling like this is important to include, even in the MVP. If it's not able to connect to the server, it's nice to have a little feedback as to why things aren't working. It doesn't have to be anything fancy. Just an 'oops, can't reach the server" or 'something went wrong' would suffice. 

EDIT: But creating those error message/pages could be a different ticket.

something15525 commented:
@scotte216 I think the error handling will be in the other ticket, where we add data visualization, or it should be it's own ticket altogether. I'm going to merge this if there are no objections.
